### DISCUSSION TITLE
Installation of Mastodon Server: Official Documentation

### DISCUSSION THREAD

**Discussion by ScubyG** [mastodon/mastodon#29013] [Q&A]
I've attempted to set up a Mastodon server multiple times, encountering failures each time despite following the official documentation. Specifically, I face an issue during the user creation step using the command:

$ adduser --disabled-login mastodon 

Upon executing this command with the '--disabled-login' option, I'm unable to log in as the 'mastodon' user.

I've experimented with both Debian 11 LXC (privileged within Proxmox) and a Debian 11 VM in Proxmox, encountering the same issue. 

While I could use 'useradd -m mastodon' as an alternative, I prefer adhering to the official documentation method.

Is the official documentation incorrect or is there something I am missing?

Any assistance would be appreciated, thank you.

**Comment by StrangeMinds1960** [comment]
So after -

adduser --disabled-login mastodon

What happens when you -

su - mastodon

**Reply by ScubyG** [reply]
Thank you for your reply, I get:

'this account is currently not available'

**Reply by StrangeMinds1960** [reply]
Very odd, there were no errors when you did -

adduser --disabled-login mastodon

Did it create a /home/mastodon directory (clutching at straws here)

You said you used the official documentation, I guess you mean -

https://docs.joinmastodon.org/admin/install/

These worked fine for me on Ubuntu 22

**Reply by ScubyG** [reply]
Yes, that is the documentation I have followed.

I just spun up a new ubuntu 22.04 lxc and it worked!

I have been trying all other than ubuntu 22.04 until now, hopefully this will now work with the rest of the documentation.

It is very strange that I had the issue with older Ubuntu's and Debian.

**Reply by StrangeMinds1960** [reply]
That's weird!!  Hope the rest of the install goes to plan

FYI there will be critical security patches released on Thursday at 15:00 UTC

**Reply by ScubyG** [reply]
so far so good, It is currently at the Ruby installation stage.

If this one works I will make another one for production use after the critical update

**Reply by ScubyG** [reply]
it worked great right up to the point of running the setup config where I then received a ton of errors:

`standard_exception_h
andling'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/commands/rake/rake_command.rb:24:in `bl
ock in perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13
.0.6/lib/rake/rake_module.rb:59:in `with_application'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/commands/rake/rake_command.rb:18:in `pe
rform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/command.rb:51:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railtie
s-7.0.8/lib/rails/commands.rb:18:in `<main>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/bootsna
p-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_req
uire.rb:32:in `require'
Tasks: TOP => db:setup => db:create
(See full trace by running task with --trace)
That failed! Perhaps your configuration is not right

The final step is compiling CSS/JS assets.
This may take a while and consume a lot of RAM.
Compile the assets now? (Y/n)

**Comment by StrangeMinds1960** [comment]
What exact command do you mean by 'running the setup config'?

This is starting to get beyond my knowledge I'm afraid

Also, I guess you didn't miss any steps? (easily done)

**Reply by ScubyG** [reply]
Thank you, I have gotten a little further than I did before, the command to run the set up is this one:

RAILS_ENV=production bundle exec rake mastodon:setup

I have a feeling from reading through the log that it may be due to postgres using UTF8 formatting so trying to find out how to create a new database with the correct encoding to see if that works, failing that I could create a separate database server for mastodon to use

**Reply by ScubyG** [reply]
I have just got Mastodon up and running on a Debian 11 LXC within Proxmox!

It turns out that the spew of errors all relate to PostgreSQL database encoding not being UTF-8, with LXC's.

I had to manually drop the mastodon database, convert the template1 database to UTF-8, create a new mastodon database using template1 and it was showing at UTF-8.

The installation then worked absolutely fine other than the missing directory `/etc/apt/keyrings/` which I had to make manually.

The other issue after that was accessing the server url via my reverse proxy server, eventually I got the right configuration and now all seems to be working well.

I am going to create a new Mastodon server on Debian 12 and maybe Alpine Linux then use one for production... as long as there are no issues with the test server I currently have

**Comment by Umiamz** [comment]
I've just installed Mastodon 4.3.0-alpha.1 on a brand new Debian 12 install. I had similar issues after creating the mastodon user but just edited /etc/passwd (with vipw) to set the shell to /bin/bash.

Also, /home/mastodon was created with 700 permissions and needed changing to 755.

I didn't have any further problems.

**Reply by ScubyG** [reply]
Thank you, maybe it is the permission settings that causes some of the issues with Mastodon, I will have a look and change the permission values to see if it resolves it on the current release
